### PR TITLE
[ALLUXIO-1759] Code refactor and documentation improvement for client-side security

### PR DESCRIPTION
**Pull request by apc999**
https://alluxio.atlassian.net/browse/ALLUXIO-1759

The goal of this PR is to simplify the code structure and make naming more consistent.

### CODE REVIEW

**Comment by apc999** [comment]
@dongche  @sundapeng, would you please take a look? 

Thanks

**Comment by dongche** [comment]
@apc999 , Thanks for the improvement! I will look at it tomorrow.

File: core/common/src/main/java/alluxio/security/authentication/NoSaslTransportProvider.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import org.apache.thrift.transport.TTransport;
+import org.apache.thrift.transport.TTransportFactory;
+
+import java.net.InetSocketAddress;
+
+import javax.annotation.concurrent.ThreadSafe;
+import javax.security.sasl.SaslException;
+
+/**
+ * If authentication type is {@link AuthType#NOSASL), be default we use this transport provider
```
Comment (dongche) [inline core/common/src/main/java/alluxio/security/authentication/NoSaslTransportProvider.java]: `be default` -> `by default`, or just remove it?

File: core/common/src/main/java/alluxio/security/authentication/TransportProviderUtils.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import org.apache.thrift.transport.TSocket;
+
+import java.net.InetSocketAddress;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * This class provides factory methods for authentication in Alluxio. Based on different
+ * authentication types specified in Alluxio configuration, it provides corresponding Thrift class
+ * for authenticated connection between Client and Server.
```
Comment (dongche) [inline core/common/src/main/java/alluxio/security/authentication/TransportProviderUtils.java]: The description seems not to be related with this class. Maybe move to `TransportProvider`?
This is a utils class to provide raw TSocket.

File: core/common/src/test/java/alluxio/security/authentication/TransportProviderTest.java [hunk 3]
```diff
@@ -34,18 +33,18 @@
 import javax.security.sasl.SaslException;
 
 /**
- * Unit test for methods of {@link AuthenticationUtils}
+ * Unit test for methods of {@link TransportProviderUtils}
```
Comment (dongche) [inline core/common/src/test/java/alluxio/security/authentication/TransportProviderTest.java]: `TransportProviderUtils` -> `TransportProvider`?

File: core/server/src/main/java/alluxio/worker/AlluxioWorker.java [hunk 4]
```diff
@@ -114,6 +114,9 @@ public static synchronized AlluxioWorker get() {
   /** Worker Web UI server. */
   private UIWebServer mWebServer;
 
+  /** The transport provider to create thrift client transport. */
```
Comment (dongche) [inline core/server/src/main/java/alluxio/worker/AlluxioWorker.java]: client -> server?

**Comment by dongche** [comment]
@apc999 , Thanks for the code structure and naming improvement! It looks pretty good! Left some minor comments about the doc.

File: core/common/src/main/java/alluxio/security/authentication/NoSaslTransportProvider.java (hunk 1 above)
Comment (apc999) [inline core/common/src/main/java/alluxio/security/authentication/NoSaslTransportProvider.java]: done

File: core/common/src/main/java/alluxio/security/authentication/TransportProviderUtils.java (hunk 2 above)
Comment (apc999) [inline core/common/src/main/java/alluxio/security/authentication/TransportProviderUtils.java]: done

Comment (apc999) [inline core/common/src/main/java/alluxio/security/authentication/TransportProviderUtils.java]: For now I kept this class as in general we don't want to put too many things in to interface. I fixed the description

File: core/common/src/test/java/alluxio/security/authentication/TransportProviderTest.java (hunk 3 above)
Comment (apc999) [inline core/common/src/test/java/alluxio/security/authentication/TransportProviderTest.java]: thanks for the catch. fixed

File: core/common/src/main/java/alluxio/AbstractClient.java [hunk 5]
```diff
@@ -66,6 +66,9 @@
    */
   protected long mServiceVersion;
 
+  /** Handler to the transport provider according to the authentication type. */
+  protected TransportProvider mTransportProvider;
```
Comment (yupeng9) [inline core/common/src/main/java/alluxio/AbstractClient.java]: final?

File: core/common/src/main/java/alluxio/security/authentication/NoSaslTransportProvider.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import javax.security.sasl.SaslException;
+
+/**
+ * If authentication type is {@link AuthType#NOSASL), we use this transport provider which simply
+ * uses default Thrift {@link TFramedTransport}.
+ */
+@ThreadSafe
+public final class NoSaslTransportProvider implements TransportProvider {
+  /** Timeout for socket in ms. */
+  private int mSocketTimeoutMs;
```
Comment (yupeng9) [inline core/common/src/main/java/alluxio/security/authentication/NoSaslTransportProvider.java]: final

File: core/common/src/main/java/alluxio/security/authentication/NoSaslTransportProvider.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  /** Max frame size of thrift transport in bytes. */
+  private int mThriftFrameSizeMax;
```
Comment (yupeng9) [inline core/common/src/main/java/alluxio/security/authentication/NoSaslTransportProvider.java]: final

File: core/common/src/main/java/alluxio/security/authentication/PlainSaslClientCallbackHandler.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+  private final String mPassword;
+
+  /**
+   * Constructs a new client side callback.
+   *
+   * @param userName the name of the user
+   * @param password the password
+   */
+  public PlainSaslClientCallbackHandler(String userName, String password) {
+    mUserName = userName;
```
Comment (yupeng9) [inline core/common/src/main/java/alluxio/security/authentication/PlainSaslClientCallbackHandler.java]: null validation?

File: core/common/src/main/java/alluxio/security/authentication/PlainSaslServer.java [hunk 9]
```diff
@@ -169,97 +156,39 @@ private void checkNotComplete() {
   }
 
   /**
-   * {@link PlainServerCallbackHandler} is used by the SASL mechanisms to get further information to
-   * complete the authentication. For example, a SASL mechanism might use this callback handler to
-   * do verification operation.
+   * This class is used to create an instances of {@link PlainSaslServer}. The parameter mechanism
```
Comment (yupeng9) [inline core/common/src/main/java/alluxio/security/authentication/PlainSaslServer.java]: parameter mechanism -> parameter

File: core/common/src/main/java/alluxio/security/authentication/PlainSaslServerCallbackHandler.java [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+public final class PlainSaslServerCallbackHandler implements CallbackHandler {
+  private final AuthenticationProvider mAuthenticationProvider;
+
+  /**
+   * Constructs a new callback handler.
+   *
+   * @param authenticationProvider the authentication provider used
+   */
+  public PlainSaslServerCallbackHandler(AuthenticationProvider authenticationProvider) {
+    mAuthenticationProvider = authenticationProvider;
```
Comment (yupeng9) [inline core/common/src/main/java/alluxio/security/authentication/PlainSaslServerCallbackHandler.java]: validation

File: core/common/src/main/java/alluxio/AbstractClient.java (hunk 5 above)
Comment (apc999) [inline core/common/src/main/java/alluxio/AbstractClient.java]: done

File: core/common/src/main/java/alluxio/security/authentication/NoSaslTransportProvider.java (hunk 6 above)
Comment (apc999) [inline core/common/src/main/java/alluxio/security/authentication/NoSaslTransportProvider.java]: done

File: core/common/src/main/java/alluxio/security/authentication/NoSaslTransportProvider.java (hunk 7 above)
Comment (apc999) [inline core/common/src/main/java/alluxio/security/authentication/NoSaslTransportProvider.java]: done

File: core/common/src/main/java/alluxio/security/authentication/PlainSaslClientCallbackHandler.java (hunk 8 above)
Comment (apc999) [inline core/common/src/main/java/alluxio/security/authentication/PlainSaslClientCallbackHandler.java]: they are not necessary to be non-null

**Comment by apc999** [comment]
PTAL, thanks!

**Comment by yupeng9** [comment]
LGTM

**Pull request merged**